lib/statespace_avx.h

```diff
@@ -228,6 +228,23 @@ struct StateSpaceAVX : public StateSpace<StateSpaceAVX<For>, For, float> {
     state.get()[p + 8] = im;
   }
 
+  // Does the equivalent of a += b elementwise.
+  void AddState(const State& a, const State& b) {
```

Comment:
Could you switch a and b here and in lib/statespace_basic.h and lib/statespace_sse.h to conform to the convention that input-only parameters are placed before output parameters.

Reply:
If we do this, could the arguments also be given more descriptive names (e.g. source, dest)?

Reply:
Changed names and order.